src/compiler/scala/tools/nsc/transform/Mixin.scala

```diff
@@ -19,8 +19,34 @@ abstract class Mixin extends InfoTransform with ast.TreeDSL {
   /** The name of the phase: */
   val phaseName: String = "mixin"
 
-  /** The phase might set the following new flags: */
-  override def phaseNewFlags: Long = lateMODULE | notOVERRIDE
+  /** Some trait methods need to be implemented in subclasses, so they cannot be private.
+    *
+    * We used to publicize during explicitouter (for some reason), so the condition is a bit more involved now it's done here
+    * (need to exclude lambdaLIFTED methods, as they do no exist during explicitouter and thus did not need to be excluded...)
+    *
+    * They may be protected, now that traits are compiled 1:1 to interfaces.
+    * (The same disclaimers about mapping Scala's notion of visibility to Java's apply.)
```

Comment:
Might be worth clarifying that despite carrying being able to answer true to `hasFlag(PROTECTED)` for longer, we are unable to emit trait methods with `ACC_PROTECTED` in bytecode due to the JVM spec:

> Methods of interfaces may have any of the flags in Table 4.6-A set except ACC_PROTECTED, ACC_FINAL, ACC_SYNCHRONIZED, and ACC_NATIVE (JLS §9.4).